src/CSnakes.Runtime/Python/PyObject.cs

```diff
@@ -279,12 +279,23 @@ public override string ToString()
         }
     }
 
-    public T As<T>() => (T)(td.ConvertTo(this, typeof(T)) ?? default!);
+    public T As<T>()
+    {
+        using (GIL.Acquire())
+        {
+            return (T)(td.ConvertTo(this, typeof(T)) ?? default!);
+        }
+    }
 
-    public static PyObject? From<T>(T value) => 
-        value is null ?
-            new PyObject(CPythonAPI.GetNone()) :
-            (PyObject?)td.ConvertFrom(value);
+    public static PyObject? From<T>(T value)
+    {
+        using (GIL.Acquire())
+        {
+            return value is null ?
+                new PyObject(CPythonAPI.GetNone()) :

```

Comment:
`new PyObject(CPythonAPI.GetNone())` seems like it could be a common type to pass around. Perhaps subclass `PyObject` and override the implementation so it never frees anything? Then you can have a `static` version for all?

```csharp
class PyObject ...
{
    internal static PyObject None => new PyObjectNone();
    ...
}
```

Reply:
I want to add a method `IsNone()` to PyObject as that's a common scenario, will include that in the PR. True and False are also Immortal objects (depending on the version of Python how that works)